#3206 Fixed escaping fields in Event handlers, Users profiles, watch lists

Patrykb0802 opened the pull request:
**Summary**
Harden XSS escaping in **Event Handlers**, **Users Profiles**, and **Watch Lists**.

**What’s changed**

* Replace unsafe string-built HTML with safe rendering:

  * Use `textContent`/dedicated spans instead of concatenating into `innerHTML`.
  * Escape watch list names in `usersProfiles.jsp`.
  * Adjust Dojo `TreeNode` titles to insert text via a safe `<span>` (no HTML injection).
* Normalize client rendering of pre-escaped values (where needed) while keeping DOM writes text-safe.
* Add `xssBean` converter for `UsersProfileVO` in `dwr.xml` to sanitize DWR payloads.
* Update `updateUserProfile(...)` to assign via `textContent`.

**Why**
Prevents script injection and DOM clobbering in user-controlled fields across these views.

**Testing**

* Names/descriptions containing `<script>` render as literal text (no execution).
* No console errors while switching profiles and opening event handler trees.
* Visual regressions: none expected.

Patrykb0802 committed:
#3206 Fixed escaping fields in Event handlers, Users profiles, watch lists

Limraj commented on WebContent/WEB-INF/jsp/usersProfiles.jsp:
```diff
@@ -110,7 +110,7 @@
 	           for (i=0; i<views.length; i++) {
 	        	   id = views[i].id;
-	               vwhtml += '<label for="vvwist'+ id +'"> '+ views[i].name +'</label><br/>';
+	               vwhtml += '<label for="vvwist'+ id +'"> '+ escapeHtml(views[i].name) +'</label><br/>';
```
Here we would need to add a converter, similarly to what we did in other places, we described it in detail in the main task.

Limraj commented on WebContent/WEB-INF/jsp/watchList.jsp:
```diff
@@ -120,14 +120,16 @@
+          const safe = document.createElement('span');
+          safe.textContent = unescapeHtml(folder.name);
           var folderNode = dojo.widget.createWidget("TreeNode", {
-                  title: "<img src='images/folder_brick.png'/> "+ folder.name,
```
Try setting:
`"<img src='images/folder_brick.png'/> <span>" + folder.name + "</span>"`
instead:
```
const safe = document.createElement('span');
safe.textContent = unescapeHtml(folder.name);
```
Moreover, in this context we use **let**, not **const**. I already mentioned this in the review on the main branch of this issue, I'm doing it again.

Limraj commented on src/br/org/scadabr/vo/usersProfiles/UsersProfileVO.java:
```diff
@@ -14,13 +14,14 @@
 
-	@JsonRemoteProperty
+	@JsonRemoteProperty @NoEscape
 	private String name;
```
As I mentioned, such data must be escaped. Everything returned from the backend must be escaped. The exclude is JavaScript scripts get from the backend, which must be executed by the client, in the browser. These scripts cannot be escaped. This applies, for example, to HTML components in graphical views. This should not apply to scripts executed on the server side, where we deliver unescaped scripts.

Limraj commented on WebContent/WEB-INF/jsp/eventHandlers.jsp:
```diff
@@ -206,11 +206,12 @@
     function createEventTypeNode(widgetId, eventType, parent) {
         var node = dojo.widget.createWidget("TreeNode", {
-                title: "<img id='"+ widgetId +"Img'/> "+ eventType.description,
+                title: "<img id='" + widgetId + "Img'/> <span id='" + widgetId + "Txt'></span>",
```
`"<img id='" + widgetId + "Img'/> <span>" + eventType.description + "</span>"`

Patrykb0802 committed:
#3206 Fixed escaping fields in Event handlers, Users profiles, watch lists

Patrykb0802 committed:
Merge remote-tracking branch 'origin/fix/#3167_Fixed_escaping_fields' into fix/#3206_Fixed_escaping_fields_in_Event_handlers_Users_profiles_watch_lists

Patrykb0802 committed:
#3206 Fixed escaping fields in Event handlers, Users profiles, watch lists

Patrykb0802 committed:
#3209 Fixed escaping fields in mailing lists, maintenance events, point links, publishers

Patrykb0802 committed:
#3206 Fixed escaping fields in Event handlers, Users profiles, watch lists

Patrykb0802 committed:
#3206 Fixed escaping fields in Event handlers, Users profiles, watch lists

Patrykb0802 committed:
#3206 Fixed escaping fields in Event handlers, Users profiles, watch lists

Patrykb0802 committed:
#3206 Fixed escaping fields in Event handlers, Users profiles, watch lists

Limraj committed:
Revert "#3206 Fixed escaping fields in Event handlers, Users profiles, watch lists"

This reverts commit 2b2db635f8cfae8f568b960476219fa0e17b81b0.

Limraj committed:
Revert "#3206 Fixed escaping fields in Event handlers, Users profiles, watch lists"

This reverts commit 44a57fcefaa58f20300e3ea45a6be94e7963db85.

Limraj committed:
Merge remote-tracking branch 'origin/fix/#3167_Fixed_escaping_fields' into fix/#3206_Fixed_escaping_fields_in_Event_handlers_Users_profiles_watch_lists

Limraj committed:
Merge remote-tracking branch 'origin/fix/#3167_Fixed_escaping_fields' into fix/#3206_Fixed_escaping_fields_in_Event_handlers_Users_profiles_watch_lists

# Conflicts:
#	WebContent/WEB-INF/dwr.xml

Limraj committed:
#3206 Fixed escaping fields in Event handlers, Users profiles, watch lists:
- Added XssLocalizableMessageConverter;

Limraj committed:
#3206 Fixed escaping fields in Event handlers, Users profiles, watch lists:
- Set bean converter for package com.serotonin.mango.web.dwr.beans;

Limraj committed:
Merge remote-tracking branch 'origin/fix/#3167_Fixed_escaping_fields' into fix/#3206_Fixed_escaping_fields_in_Event_handlers_Users_profiles_watch_lists

# Conflicts:
#	WebContent/WEB-INF/dwr.xml

Limraj committed:
#3206 Fixed escaping fields in Event handlers, Users profiles, watch lists:
- Set bean converter for package com.serotonin.mango.vo.event.handlers;

Limraj committed:
#3206 Fixed escaping fields in Event handlers, Users profiles, watch lists:
- Set bean converter for com.serotonin.mango.vo.publish.PublisherVO;

Patrykb0802 committed:
#3206 Fixed escaping fields in Event handlers, Users profiles, watch lists

Limraj committed:
Merge remote-tracking branch 'origin/fix/#3167_Fixed_escaping_fields' into fix/#3206_Fixed_escaping_fields_in_Event_handlers_Users_profiles_watch_lists

Limraj committed:
Merge remote-tracking branch 'origin/fix/#3167_Fixed_escaping_fields' into fix/#3206_Fixed_escaping_fields_in_Event_handlers_Users_profiles_watch_lists

Limraj committed:
#3206 Fixed escaping fields in Event handlers, Users profiles, watch lists:
- Corrected XssLocalizableMessageConverter;

Limraj committed:
#3206 Fixed escaping fields in Event handlers, Users profiles, watch lists:
- Set bean converter for package com.serotonin.mango.vo.report in dwr.xml;

Limraj commented on WebContent/WEB-INF/dwr.xml:
```diff
@@ -129,7 +129,7 @@
     <convert converter="bean" match="com.serotonin.mango.view.ShareUser"/>
 
-    <convert converter="bean" match="com.serotonin.mango.vo.bean.*"/>
+    <convert converter="xssBean" match="com.serotonin.mango.vo.bean.*"/>
```
`<convert converter="xssBean" match="com.serotonin.mango.vo.bean.*"/>`
replace with:
`<convert converter="bean" match="com.serotonin.mango.vo.bean.*"/>`
and convert only selected specific types from this package, e.g.:
`<convert converter="xssBean" match="com.serotonin.mango.vo.bean.PointHistoryCount"/>`
...

Limraj commented on WebContent/WEB-INF/jsp/watchList.jsp:
```diff
@@ -154,7 +154,7 @@
           var spanNode = document.createElement("span");
           spanNode.id = 'ph'+ point.key +'Name';
-          spanNode.textContent = point.value;
+          spanNode.textContent = unescapeHtml(point.value);
```
`spanNode.textContent = unescapeHtml(point.value);`
replace with:
`spanNode.innerHTML = point.value;`

Limraj committed:
Merge remote-tracking branch 'origin/fix/#3167_Fixed_escaping_fields' into fix/#3206_Fixed_escaping_fields_in_Event_handlers_Users_profiles_watch_lists

Patrykb0802 committed:
#3206 Fixed escaping fields in Event handlers, Users profiles, watch lists

Patrykb0802 committed:
#3206 Fixed escaping fields in Event handlers, Users profiles, watch lists

Patrykb0802 committed:
Revert "#3206 Fixed escaping fields in Event handlers, Users profiles, watch lists"

This reverts commit 371ad447775f6289a09d769b260ed6cd5e32cf19.

Limraj committed:
#3206 Fixed escaping fields in Event handlers, Users profiles, watch lists:
- Fixed double escaped alias in: MaintenanceEventVO, ScheduledEventVO (use NoEscape annotation);
- Fixed escaped after save event handler;

Limraj committed:
#3206 Fixed escaping fields in Event handlers, Users profiles, watch lists:
- Removed use NoEscape annotation in: MaintenanceEventVO.alias, ScheduledEventVO.alias
- Unescaped in getDescription string fields from: MaintenanceEventVO, ScheduledEventVO because returned LocalizableMessage object escaped by converter;
- Corrected maintenanceEvents.jsp, scheduledEvents.jsp use unescapeHtml;

Limraj committed:
#3206 Fixed escaping fields in Event handlers, Users profiles, watch lists:
- Corrected: MaintenanceEventVO.getDescription, ScheduledEventVO.getDescription;

Limraj reviewed (approved):
handlers works:
<img width="975" height="904" alt="image" src="https://github.com/user-attachments/assets/cb08a5db-f2ee-440f-aaa1-b60ba2913efd" />

user profiles works:
<img width="1159" height="733" alt="image" src="https://github.com/user-attachments/assets/0d166f84-e278-4d61-94f9-200f3ea41b0e" />

watch lists:
<img width="1918" height="367" alt="image" src="https://github.com/user-attachments/assets/5c4591d1-0a34-4332-933f-3d6fb0cc826e" />

Event Scheduled works:
<img width="570" height="381" alt="image" src="https://github.com/user-attachments/assets/1ae5926f-4ad2-4c0d-ada8-886c6ed19338" />

Maintenance Event works:
<img width="632" height="261" alt="image" src="https://github.com/user-attachments/assets/59826dbd-c489-4920-82e7-479b2eb3b6fb" />
